src/StakeRegistry.sol

```diff
@@ -418,6 +418,7 @@ contract StakeRegistry is StakeRegistryStorage {
                 "StakeRegistry._addStrategyParams: cannot add strategy with zero weight"
             );
             strategyParams[quorumNumber].push(_strategyParams[i]);
+            strategiesPerQuorum[quorumNumber].push(_strategyParams[i].strategy);
```

Comment:
Well, I wish we weren't duplicating storage here but... this is a much simpler change than splitting `strategyParams` into 2 arrays.

And I assume we have backend stuff relying on `strategyParams` and at this point we'd like to not change that.

Reply:
oof yes agreed this is ugly but I can see the reasoning.
will need to add other parallel storage updates to make these 2 lists consistent, as right now this PR appears to just be duplicating adding entries (but not removing them).

one option could be to split `strategyParams` into 2 arrays, but support a view function which combines an entry from each array to maintain interface compatibility -- but I assume this still would break indexing, so...worst of both worlds perhaps.

Reply:
> will need to add other parallel storage updates to make these 2 lists consistent

pulling this out -- `removeStrategies` needs to delete entries from this list

Reply:
Yeah as this was a last minute optimization onchain, I figured we shouldn't change the current storage layout here. Perhaps the next testnet deployment this could be updated?

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/170/commits/3688a02f4a7841f2d3befbdd3821cc3b4d22c602#diff-ff6dec6ce26deefc6d5cb801b1d654b03cba3bd352386f0a828990d979a97332R241
Fixed here for removing strategies